core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/TupleCodec.java

```diff
@@ -186,15 +187,25 @@ public TupleValue parse(@Nullable String value) {
       } catch (IllegalArgumentException e) {
         throw new IllegalArgumentException(
             String.format(
-                "Cannot parse tuple value from \"%s\", invalid CQL value at character %d",
-                value, position),
+                "Cannot parse tuple value from \"%s\", invalid CQL value at character %d: %s",
+                value, position, e.getMessage()),
             e);
       }
 
       String fieldValue = value.substring(position, n);
       DataType elementType = cqlType.getComponentTypes().get(i);
       TypeCodec<Object> codec = registry.codecFor(elementType);
-      tuple = tuple.set(i, codec.parse(fieldValue), codec);
+      Object parsed;
+      try {
+        parsed = codec.parse(fieldValue);
+      } catch (Exception e) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Cannot parse tuple value from \"%s\", invalid CQL value at character %d: %s",
+                value, position, e.getMessage()),
+            e);
```

Comment:
This can be extracted to some method because the logic of constructing this message is duplicated in a couple places

Reply:
There are only 2 occurrences of this exact message, the others are slightly different, so I think it's not worth it.